docs/Design/Null Check.md

```diff
@@ -0,0 +1,140 @@
+# Null Check
+Creates checks for if the passed variable has value null and/or undefined (see errata). Can be used to check positvly for null or negetivly for null
+
+## Commands
+
+### `is null`
+
+`is null : variable`
+
+Creates a test that will return true if the variable is NULL
+
+### `not null`
+
+`not null : variable`
+
+Creates a test that will return false if the variable is NULL
+
+## Usage
+
+```
+is null : foo
+```
+```
+not null : bar
+```
+
+### CSharp
+
+```csharp
+foo == null
+```
+```csharp
+bar != null
+```
+
+### Java
+
+```java
+foo == null
+```
+```java
+bar != null
+```
+
+### Python
+
+```python
+foo is None
+```
+```python
+bar is not None
+```
+
+### Ruby
+
+```ruby
+foo.nil?
+```
+```ruby
+!bar.nil?
+```
+
+### Typescript
+
+```javascript
+foo == null
+```
+```javascript
+foo != null
+```
+
+## Implementation
+
+A null check can either test for a variable being null, or test for a variable not being null.

```

Comment:
This sentence is redundant (we already understand that there are two commands) and not actually related to the implementation.